tests/Engine.Tests/GameEngineTests.cs

```diff
@@ -42,6 +42,38 @@ public void WhenConfigDoesNotProvideOptionsThenThrowsException()
             }
         }
 
+        public class Dispose
+        {
+            [UnitTest]
+            public void WhenDisposedThenShouldThrowObjectDisposedExceptionWhenPublish()
+            {
+                // TODO(DANjEEEEE): I'm not sure if this is the correct way to test / implement this, so any advice would be useful.
+                var engine = new GameEngine<Fake.GameTime>(DefaultConfig, DefaultTranslator, DefaultClock);
+                using (engine) { }
+
+                Action act = () => engine.Publish(new Fake.Event());
+                act.ShouldThrow<ObjectDisposedException>();
+            }
+
+            [UnitTest]
+            public void WhenDisposedThenShouldThrowObjectDisposedExceptionWhenSubscribe()
+            {
+                // TODO(DANjEEEEE): I'm not sure if this is the correct way to test / implement this, so any advice would be useful.
+                var engine = new GameEngine<Fake.GameTime>(DefaultConfig, DefaultTranslator, DefaultClock);
+                var observer = new Mock<IObserver<Fake.Event>>();
+                using (engine) { }
+
+                Action act = () => engine.Subscribe(observer.Object);
+                act.ShouldThrow<ObjectDisposedException>();
+            }
+
+            [UnitTest]
+            public void WhenDisposedThenSubscriptionsShouldAlsoBeDisposed()
+            {
+                // TODO(DANjEEEEE): I'm not sure how I should have written a test for this, so any advice would be useful.
```

Comment:
Will write up so code and PR to you when I get a chance for this test, but for now make sure the test is failing (an empty test is a passed test). (`Assert.True(false)` or some other dummy logic).